crypto/slh_dsa/slh_dsa.c

```diff
@@ -348,7 +351,7 @@ static int get_tree_ids(PACKET *rpkt, const SLH_DSA_PARAMS *params,
      *     when X = 54 it would be A & (0x3F_FFFF_FFFF_FFFF)
      * i.e. A & (0xFFFF_FFFF_FFFF_FFFF >> (64 - X))
      */
-    tree_id_mask = ((uint64_t)-1) >> (64 - (params->h - params->hm));
+    tree_id_mask = ((uint64_t)~0) >> (64 - (params->h - params->hm));
```

Comment:
The `~` may need to go in front of the cast, otherwise, this could be a cast from the (int)~0 to 64 bits, which isn't what you want. I wrote the `~` that way deliberately in the suggestion.